sqlmesh/core/model/meta.py

```diff
@@ -53,6 +54,9 @@ class ModelMeta(_Node):
 
     dialect: str = ""
     name: str
+    on_destructive_change_: t.Optional[OnDestructiveChange] = Field(
```

Comment:
Why do we have this both here and in the model kind?

Reply:
There's two ways for a user to specify this value: in a model definition and in the model defaults.

If they specify in the definition, it's captured when we parse that definition and available to the kind constructor in the _model_kind_validator.

If they specify in the model defaults, this is the path for that value to get to the kind constructor in the _model_kind_validator.

Reply:
I feel like we need to find a different way to propagate the default. The current way looks more like a workaround.

Reply:
I think it's fair to say it's a workaround for pydantic field validators only having access to fields prior to them in the pydantic model def.